mutation_partition.cc

```diff
@@ -250,15 +250,33 @@ void mutation_partition::ensure_last_dummy(const schema& s) {
 }
 
 void mutation_partition::apply(const schema& s, const mutation_partition& p, const schema& p_schema) {
-    apply_weak(s, p, p_schema);
+    mutation_application_stats app_stats;
+    return apply(s, p, p_schema, app_stats);
```

Comment:
I'm not sure I'm following: where those app_stats eventually go after apply() updates them?

Reply:
And C++ wise the above looks weird at least (it's unclear how this even compiles): you have a return statement in the function returning `void`.

Reply:
@vladzcloudius it's legal to return void (very useful for templates). But should be when not needed, it's surprising.